File: src/phase.py
from src.database import get_connection


def create_phase(project_id, name, description, order_index, status="TODO"):
    connection = get_connection()
    cursor = connection.cursor()

    cursor.execute("""
        INSERT INTO phase (
            project_id,
            name,
            description,
            order_index,
            status,
            created_at,
            updated_at
        )
        VALUES (?, ?, ?, ?, ?, datetime('now'), datetime('now'))
    """, (
        project_id,
        name,
        description,
        order_index,
        status
    ))

    connection.commit()

    phase_id = cursor.lastrowid

    connection.close()

    return phase_id


def get_phase(phase_id):
    connection = get_connection()
    cursor = connection.cursor()

    cursor.execute("""
        SELECT
            id,
            project_id,
            name,
            description,
            order_index,
            status,
            created_at,
            updated_at
        FROM phase
        WHERE id = ?
    """, (phase_id,))

    phase = cursor.fetchone()

    connection.close()

    return phase


def get_project_phases(project_id):
    connection = get_connection()
    cursor = connection.cursor()

    cursor.execute("""
        SELECT
            id,
            project_id,
            name,
            description,
            order_index,
            status,
            created_at,
            updated_at
        FROM phase
        WHERE project_id = ?
        ORDER BY order_index
    """, (project_id,))

    phases = cursor.fetchall()

    connection.close()

    return phases

File: src/models/Phase.ts
import { query } from '../config/database.js';
import { IPhase, IPhaseCreateInput } from './types.js';

export class PhaseModel {
  static async create(input: IPhaseCreateInput): Promise<IPhase> {
    const res = await query(
      `INSERT INTO phases (project_id, name, description, order_index, status)
       VALUES ($1, $2, $3, $4, $5)
       RETURNING *`,
      [
        input.project_id,
        input.name,
        input.description ?? null,
        input.order_index ?? 0,
        input.status ?? 'PLANNED',
      ]
    );
    return res.rows[0];
  }

  static async findById(id: string): Promise<IPhase | null> {
    const res = await query('SELECT * FROM phases WHERE id = $1', [id]);
    return res.rows[0] || null;
  }

  static async findByProjectId(projectId: string): Promise<IPhase[]> {
    const res = await query(
      'SELECT * FROM phases WHERE project_id = $1 ORDER BY order_index ASC, created_at ASC',
      [projectId]
    );
    return res.rows;
  }

  static async deleteById(id: string): Promise<boolean> {
    const res = await query('DELETE FROM phases WHERE id = $1', [id]);
    return (res.rowCount ?? 0) > 0;
  }
}
